Core/src/org/sleuthkit/autopsy/datamodel/DataModelActionsFactory.java

```diff
@@ -64,129 +71,227 @@
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(ExtractAction.getInstance());
+        actionsList.add(new HashSearchAction(SEARCH_FOR_FILES_SAME_MD5, fileNode));
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(AddContentTagAction.getInstance());
         if (isArtifactSource) {
-            actions.add(AddBlackboardArtifactTagAction.getInstance());
+            actionsList.add(AddBlackboardArtifactTagAction.getInstance());
         }
-        actions.addAll(ContextMenuExtensionPoint.getActions());
-        return actions;
+        
+        final Collection<AbstractFile> selectedFilesList =
+                new HashSet<>(Utilities.actionsGlobalContext().lookupAll(AbstractFile.class));
+        if(selectedFilesList.size() == 1) {
+            actionsList.add(DeleteFileContentTagAction.getInstance());
+        }
+        if(isArtifactSource) {
+            final Collection<BlackboardArtifact> selectedArtifactsList =
+                    new HashSet<>(Utilities.actionsGlobalContext().lookupAll(BlackboardArtifact.class));
+            if(selectedArtifactsList.size() == 1) {
+                actionsList.add(DeleteFileBlackboardArtifactTagAction.getInstance());
+            }
+        }
+        
+        actionsList.addAll(ContextMenuExtensionPoint.getActions());
+        return actionsList;
     }
     
     public static List<Action> getActions(SlackFile slackFile, boolean isArtifactSource) {
-        List<Action> actions = new ArrayList<>();
-        actions.add(new ViewContextAction((isArtifactSource ? VIEW_SOURCE_FILE_IN_DIR : VIEW_FILE_IN_DIR), slackFile));
+        List<Action> actionsList = new ArrayList<>();
+        actionsList.add(new ViewContextAction((isArtifactSource ? VIEW_SOURCE_FILE_IN_DIR : VIEW_FILE_IN_DIR), slackFile));
         final SlackFileNode slackFileNode = new SlackFileNode(slackFile);
-        actions.add(null); // creates a menu separator
-        actions.add(new NewWindowViewAction(VIEW_IN_NEW_WINDOW, slackFileNode));
-        actions.add(null); // creates a menu separator
-        actions.add(ExtractAction.getInstance());
-        actions.add(null); // creates a menu separator
-        actions.add(AddContentTagAction.getInstance());
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(new NewWindowViewAction(VIEW_IN_NEW_WINDOW, slackFileNode));
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(ExtractAction.getInstance());
+        actionsList.add(null); // creates a menu separator
+        actionsList.add(AddContentTagAction.getInstance());
         if (isArtifactSource) {
-            actions.add(AddBlackboardArtifactTagAction.getInstance());
+            actionsList.add(AddBlackboardArtifactTagAction.getInstance());
+        }
+        
+        final Collection<AbstractFile> selectedFilesList =
```

Comment:
Condensing these blocks of code into something like  
if(Utilities.actionsGlobalContext().lookupAll(AbstractFile.class).size() == 1){
 actionsList.add(DeleteFileContentTagAction.getInstance());
}
Would be possible, I don't know if it is more or less readable though so probably a matter of preference.